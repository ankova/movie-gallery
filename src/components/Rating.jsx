import React, { useState, Fragment } from 'react';


const Rating = (props) => {
    const {onChange} = props;
    const [value, setValue] = useState(3);
    
    const handleChange = (e) => {
        setValue(e.target.value);
        onChange(e.target.value)
    }

    return (
        <div>
            <label htmlFor="rating">Rating ({value})</label>
            <input type="range" id="rating" name="cowbell" 
                min="0" max="10" defaultValue="3" step="0.5" onChange={handleChange}/>
        </div>
    )
}

export default Rating;