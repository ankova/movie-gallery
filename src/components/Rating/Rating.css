.Raiting {
  margin-top: 2em;
  display: inline-block;
  width: min-content; }
  .Raiting input[type="range" i] {
    cursor: grab;
    vertical-align: bottom;
    margin-top: 1em; }
