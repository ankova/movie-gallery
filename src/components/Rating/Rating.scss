.Raiting {
    margin-top: 2em;
    display: inline-block;
    width: min-content;
  
    input[type="range" i] {
      cursor: grab;
      vertical-align: bottom;
      margin-top: 1em;
    }
  }