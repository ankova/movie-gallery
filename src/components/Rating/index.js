import Raiting from './Rating';

export default Raiting;