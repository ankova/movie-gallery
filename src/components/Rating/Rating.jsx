import React, { useState } from 'react';

import './Rating.css';


const Rating = (props) => {
    const {onChange} = props;
    const [value, setValue] = useState(3);
    
    const handleChange = (e) => {
        setValue(e.target.value);
        onChange(e.target.value)
    }

    return (
        <div className="Raiting">
            <label htmlFor="rating">Sort by rating {value}</label>
            <input type="range" id="rating" name="cowbell" 
                min="0" max="10" defaultValue="3" step="0.5" onChange={handleChange}/>
        </div>
    )
}

export default Rating;