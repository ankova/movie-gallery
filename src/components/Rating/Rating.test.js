import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Rating from './Rating';


describe('Rating', () => {
    let component;
    const handleChange = jest.fn();
    const props = {
        onChange: handleChange
    };

    it('renders component without crashing', () => {
        const tree = renderer.create(<Rating />).toJSON;
        expect(tree).toMatchSnapshot();
    });

    beforeEach(() => component = mount(<Rating {...props} />));

    it('checks if setValue works', () => {
        const event = {
            target: {value: 5}
        };

        expect(component.find('label').text()).toBe('Sort by rating 3');

        component.find('input').simulate('change', event);
        expect(handleChange).toHaveBeenCalledWith(5);
        expect(component.find('label').text()).toBe('Sort by rating 5');
    });
});
