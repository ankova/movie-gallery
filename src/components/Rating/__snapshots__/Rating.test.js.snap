exports[`Rating renders component without crashing 1`] = `[Function]`;
